lib/framework/framework.dart

```diff
@@ -0,0 +1,289 @@
+      CoreElement e = new CoreElement.from(element);
+      bool isCurrent = current.ref == element.attributes['href'];
+      e.toggleClass('active', isCurrent);
+    }
+
+    // status
+    CoreElement helpLink = new CoreElement.from(querySelector('#docsLink'));
+    HelpInfo helpInfo = current.helpInfo;
+    if (helpInfo == null) {
+      helpLink.hidden(true);
+    } else {
+      helpLink.clear();
+      helpLink.add([
+        span(text: '${helpInfo.title} '),
+        span(c: 'octicon octicon-link-external small-octicon'),
+      ]);
+      helpLink.setAttribute('href', helpInfo.url);
+      helpLink.hidden(false);
+    }
+  }
+
+  void showError(String title, [dynamic error]) {
+    PFlash flash = new PFlash();
+    flash.addClose().click(clearError);
+    flash.add(span(text: title));
+    if (error != null) {
+      flash.add(new CoreElement('br'));
+      flash.add(span(text: '$error'));
+    }
+
+    CoreElement errorContainer =
+        new CoreElement.from(querySelector('#error-container'));
+    errorContainer.add(flash);
+  }
+
+  void clearError() {
+    querySelector('#error-container').children.clear();
+  }
+}
+
+class StatusLine {
+  final CoreElement element;
+  final List<StatusItem> _items = [];
+
+  StatusLine(this.element);
+
+  void add(StatusItem item) {
+    _items.add(item);
+
+    _rebuild();
+  }
+
+  void _rebuild() {
+    element.clear();
+
+    if (_items.isNotEmpty) {
+      element.add(_items.first.element);
+
+      for (StatusItem item in _items.sublist(1)) {
+        element.add(new SpanElement()
+          ..text = '•'
+          ..classes.add('separator'));
+        element.add(item.element);
+      }
+    }
+  }
+
+  void remove(StatusItem item) {
+    _items.remove(item);
+
+    _rebuild();
+  }
+
+  void addAll(List<StatusItem> items) {
+    _items.addAll(items);
+
+    _rebuild();
+  }
+
+  void removeAll() {
+    _items.clear();
+    _rebuild();
+  }
+}
+
+void toast(String message) {
+  // TODO:
+  print(message);
+}
+
+abstract class Screen {
+  final String name;
+  final String id;
+  final String iconClass;
+
+  final Property<bool> _visible = new Property(true);
+
+  final List<StatusItem> statusItems = [];
+
+  Screen(this.name, this.id, [this.iconClass]);
+
+  String get ref => id == '/' ? id : '/$id';
+
+  bool get visible => _visible.value;
+
+  set visible(bool value) {
+    _visible.value = value;
+  }
+
+  Stream<bool> get onVisibleChange => _visible.onValueChange;
+
+  void createContent(Framework framework, CoreElement mainDiv) {}
+
+  void entering() {}
+
+  void exiting() {}
+
+  // TODO: generalize this - global and page status items
+  void addStatusItem(StatusItem item) {
+    // TODO: If we're live, add to the screen
+    statusItems.add(item);
+  }
+
+  void removeStatusItems(StatusItem item) {
+    // TODO: If we're live, remove from the screen
+    statusItems.remove(item);
+  }
+
+  HelpInfo get helpInfo => null;
+
+  String toString() => id;
```

Comment:
is there a reason this toString shouldn't be
=> 'Screen($id)'
to disambiguate?